src/System.Net.Security/src/System/Net/Security/SslStream.cs

```diff
@@ -283,6 +283,11 @@ public virtual Task AuthenticateAsServerAsync(X509Certificate serverCertificate,
                 (callback, state) => ((SslStream)state).BeginShutdown(callback, state),
                 iar => ((SslStream)iar.AsyncState).EndShutdown(iar), 
                 this);
+
+        public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            return _sslState.SecureStream.WriteAsync(buffer, offset, count, cancellationToken);
```

Comment:
Does the wrapped stream's WriteAsync do argument validation, or does that need to be done here?

Reply:
Argument validation is done in sslstreaminternal